src/org/parosproxy/paros/extension/report/ReportGenerator.java

```diff
@@ -126,7 +129,8 @@ public static File stringToHtml(String inxml, String infilexsl, String outfilena
 					bw = new BufferedWriter(new FileWriter(outfilename + "-orig.xml"));
 					bw.write(inxml);
 				} catch (IOException e2) {
-					logger.error(e.getMessage(), e);
+					logger.error(Constant.messages.getString("report.write.diskspace.error.log.message"), e);
+					JOptionPane.showMessageDialog(View.getSingleton().getMainFrame(), Constant.messages.getString("report.write.diskspace.error.log.message"));

```

Comment:
That was a side comment, didn't mean you should fix/change that.
But if you want, yeah, it should probably return null. Some of the callers ignore the return value, but one is already showing a custom error message if the file is null. [1] That caller would have to be changed to not show the message.

[1] https://github.com/zaproxy/zaproxy/blob/c678ac66c740290df8956c835596acecc2198ed8/src/org/parosproxy/paros/extension/report/ReportLastScan.java#L190-L194